core/src/main/java/org/apache/calcite/sql/SqlTablePartitionExpression.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql;
+
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.util.ImmutableNullableList;
+
+import java.util.List;
+/**
+ * Parse tree for Partition Expression.
+ */
+public class SqlTablePartitionExpression extends SqlCall {
+  public static final SqlSpecialOperator OPERATOR =
+      new SqlSpecialOperator("PARTITION_EXPRESSION_WITH_CONSTANT",
+          SqlKind.OTHER);
+  public final SqlNode partitionExpression;
+  public final int extraNumberOfPartitions;
+
+  /**
+   * Creates a {@code SqlTablePartitionExpression}.
+   * @param pos                     Parser position, must not be null.
+   * @param partitionExpression     Partition expressions in a SqlNodeList.
+   * @param extraNumberOfPartitions Big Integer represent the extra number
```

Comment:
Seems like a regular int is passed and not a BigInteger?

Reply:
Thanks for pointing this out. When I was implementing the code and checking the parsing logic, calcite currently don't have a way to parser big integer. I should've communicate this point across in my design doc. I will note this point in my design doc now and we can implement support for parsing big integer in the future.